### PR TITLE
Lineage from Unity Catalog

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

 Extract lineage information provided by Databricks

### 🤓 What?

- Extract Table/Column lineage from Databricks API
- Use low-level `perform_query` to request the correct endpoint.
- Add tests

### 🧪 Tested?

Unit tests, ingested the MCE to latest dev environment.

### CODE REVIEW

File: metaphor/unity_catalog/extractor.py [hunk 1]
```diff
@@ -72,12 +78,12 @@ async def extract(self) -> Collection[ENTITY_TYPES]:
                     continue
 
                 for table in self._get_tables(catalog, schema):
+                    table_name = f"{catalog}.{schema}.{table.name}"
                     if not self._filter.include_table(catalog, schema, table.name):
-                        logger.info(
-                            f"Ignore table: {catalog}.{schema}.{table.name} due to filter config"
```
Comment (mars-lan) [inline metaphor/unity_catalog/extractor.py:77]: Why dropping the extra (useful) info in the log?

File: metaphor/unity_catalog/extractor.py [hunk 2]
```diff
@@ -146,6 +152,42 @@ def _init_dataset(self, table: Table):
...
 
+        return dataset
+
+    def _populate_lineage(self, dataset: Dataset):
+        table_name = f"{dataset.structure.database}.{dataset.structure.schema}.{dataset.structure.table}"
+        lineage = list_table_lineage(self._api.client.client, table_name)
+
+        # Skip table without upstream
+        if not lineage.upstreams:
+            return
```
Comment (mars-lan) [inline metaphor/unity_catalog/extractor.py:163]: Consider adding a info/debug log to make future debugging easier.

File: metaphor/unity_catalog/extractor.py (hunk 1 above)
Comment (elic-eon) [reply metaphor/unity_catalog/extractor.py:77]: change it to `table_name` here.

File: metaphor/unity_catalog/extractor.py (hunk 2 above)
Comment (elic-eon) [reply metaphor/unity_catalog/extractor.py:165]: Added

**Review by mars-lan — APPROVED** [review]

**Pull request merged**